water_mark/water_mark.py

```diff
@@ -0,0 +1,303 @@
+###############################################################################
+#########   __        __    _            __  __            _    ###############
+##########  \ \      / /_ _| |_ ___ _ __|  \/  | __ _ _ __| | __ ##############
+###########  \ \ /\ / / _` | __/ _ \ '__| |\/| |/ _` | '__| |/ /  #############
+############  \ V  V / (_| | ||  __/ |  | |  | | (_| | |  |   <    ############
+############   \_/\_/ \__,_|\__\___|_|  |_|  |_|\__,_|_|  |_|\_\  #############
+###########                                                      ##############
+##########     George Meier 15 July 2019                        ###############
+#########      ASF Water Detection AI Training Data Generator  ################
+########################################################################80chars
+# This program, given VV and VH sar images, outputs a directory of tiled and
+# labeled sar and mask images.
+#
+# Begin by setting up the file directory shown below
+# When identify_water.py runs, a gui will appear running an algorithm to
+# generate 'synthetic' water mask tifs.  Use the magnifying glass icon to zoom
+# in on the lower lefthand corner.  Click the magnifying glass agin to change
+# the curser back to normal, then click to draw a box over an area of interest.
+# Try many areas and change the bin size until the mask looks like a good fit
+# to its corresponding sar images.  When satisfied click the make mask button.
+
+###############################################################################
+# Python3
+# Windows admin powershell or Linux
+# cmd line arguments
+#   '-waterMask' '-size=<int>' to set mxm_tile_size
+# import function w/ arguments
+#   'main(waterMask=True, size=<int>)' # NOTE 512 IS NOT DEFAULT HERE
+#
+# - water_mark:
+#   - inputs:
+#       - sar.tif (1+)
+#       - identify_water.py  (neede for waterMask)
+#       - gdal_reclassify.py (needed for waterMAsk)
+#   - syntheticTriainingData<date>: (output directory made automatically)
+#       - sar: (1+)
+#           - Original and tiled VV, VH, and Mask images.
+#   - water_mark.py
+#   - etl_water_mark.py
+#   - download-all-<nums>.py (from ASF hyp3)
+###############################################################################
+
+import argparse
+import os
+import shutil
+from osgeo import gdal
+from identify_water import main as idw_main
+from datetime import date
+from typing import Tuple, Dict, List
+
+from etl_water_mark import detect_windows_OS
+from gdal_reclassify import processDataset
+
+
+def make_database() -> Dict[str, Tuple[str, str]]:
+    data_list = []
```

Comment:
Something like this should do the same thing as the following 4 lines.
```suggestion
    data_list = sorted(filter(lambda fname: fname.endswith('.tif'), os.listdir('inputs')))
```

Reply:
Or if you prefer a list comprehension

```suggestion
    data_list = sorted([fname for fname in os.listdir('inputs') if fname.endswith('.tif')])
```